src/constants/docsComponentMapping.js

```diff
@@ -179,12 +179,8 @@ const OrangeTableCell = styled.td`
 export const apiReferenceComponents = {
   ...components,
   wrapper: WrapperApiReference,
-  h1: styled(components.h1).attrs(() => ({
-    as: makeLinkedHeader(ApiRefH1, headerOptions),
-  }))``,
-  h2: styled(components.h2).attrs(() => ({
-    as: makeLinkedHeader(ApiRefH2, headerOptions),
-  }))``,
+  h1: makeLinkedHeader(ApiRefH1, headerOptions),
+  h2: makeLinkedHeader(ApiRefH2, headerOptions),
```

Comment:
Styled components' `as` prop get wibbly when there are multiple levels of components in question. Since there aren't any additional styles here, I think this is complexity with no benefit, and this appears to be how some unexpected styles are being added to the static HTML (though it's weird that it's not also applying to the client render)